modules/renter/contractor/contracts.go

```diff
@@ -416,16 +417,37 @@ func (c *Contractor) threadedContractMaintenance() {
 			// cycle. This is calculated by starting with the total cost and
 			// subtracting out all of the fees, and then all of the unused
 			// money that was allocated (the RenterFunds).
-			renewAmount := contract.TotalCost.Sub(contract.ContractFee).Sub(contract.TxnFee).Sub(contract.SiafundFee).Sub(contract.RenterFunds)
+			//
+			// In order to accurately fund contracts based on variable spending,
+			// the cost per block is calculated based on the total spent
+			// over the length of time that the contract was active before
+			// renewal.
+			oldContractSpent := contract.TotalCost.Sub(contract.ContractFee).Sub(contract.TxnFee).Sub(contract.SiafundFee).Sub(contract.RenterFunds)
+			oldContractLength := blockHeight - contract.StartHeight
+			if oldContractLength <= 1 {
+				oldContractLength = types.BlockHeight(1)
```

Comment:
since these are unsigned ints, `== 0` is probably clearer, and makes it more obvious that we're trying to avoid a divide-by-zero panic.